### PR TITLE
Fix white flash issues on new window/tab opening

### PR DESCRIPTION
**Pull request by goodov**
Fix white flash issues on new window creation (Windows-specific, see [1] for details) and on NTP content loading. NTP content loading issue seems like only reproduced on Windows because of the way Chromium widgets are wired.

<details>
<summary>Before</summary>

https://github.com/user-attachments/assets/a62fd9a9-3e57-4fb8-8709-422e9260ca0b

</details>

<details>
<summary>After</summary>

https://github.com/user-attachments/assets/05194caf-7ca8-48dd-898e-636e006c577c

</details>

1. https://stackoverflow.com/questions/69715610/how-to-initialize-the-background-color-of-win32-app-to-something-other-than-whit

Resolves https://github.com/brave/brave-browser/issues/38214

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/brave_ui_features.cc [hunk 1]
```diff
@@ -5,9 +5,20 @@
...
 BASE_FEATURE(kBraveNtpSearchWidget,
              "BraveNtpSearchWidget",
              base::FEATURE_ENABLED_BY_DEFAULT);
-}
+
+#if BUILDFLAG(IS_WIN)
+// Enables window cloaking on window creation to prevent a white flash.
+BASE_FEATURE(kBraveWorkaroundNewWindowFlash,
+             "BraveWorkaroundNewWindowFlash",
+             base::FEATURE_DISABLED_BY_DEFAULT);
```
Comment (goodov) [inline browser/ui/brave_ui_features.cc:21]: disabled by default as the change may potentially break the browser startup. Enabling the feature via Griffin is safer as Griffin has fallback mechanism to revert the seed if multiple crashes was detected at startup.

**Review by boocmp — APPROVED** [review]

**Comment by simonhong** [comment]
Hmm.. I'm still seeing the issue with this PR on my Win11

https://github.com/user-attachments/assets/bd3320f2-3c55-485c-8583-92d1f0b89ed5

**Comment by goodov** [comment]
> Hmm.. I'm still seeing the issue with this PR on my Win11

I have animations disable so I missed that :o

**Comment by simonhong** [comment]
> > Hmm.. I'm still seeing the issue with this PR on my Win11
> 
> I have animations disabled so I missed that :o

Oh, wait!! NVM. I didn't enable the flag! It works well 👍🏼

**Review by simonhong — APPROVED** [review]
++

**Comment by goodov** [comment]
> > > Hmm.. I'm still seeing the issue with this PR on my Win11
> > 
> > 
> > I have animations disabled so I missed that :o
> 
> Oh, wait!! NVM. I didn't enable the flag! It works well 👍🏼

phew! But with the flag enabled we don't see this shiny window appearance animation. I guess it's an acceptable tradeoff here.

**Review by iefremov — APPROVED** [review]

**Pull request merged**